src/ORM/Hierarchy/Hierarchy.php

```diff
@@ -418,7 +418,7 @@ private function getHierarchyBaseClass(): string
     {
         $ancestry = ClassInfo::ancestry($this->owner);
         $ancestorClass = array_shift($ancestry);
-        while ($ancestorClass && !Extensible::has_extension($ancestorClass, self::class)) {
+        while ($ancestorClass && !$ancestorClass::has_extension(self::class)) {
```

Comment:
PHP8.1 doesn't want you to call static method on traits anymore. `Extensible::has_extension()` was the biggest offender here.